lib/chef/resource.rb

```diff
@@ -138,8 +138,11 @@ def initialize(name, run_context=nil)
     #
```

Comment:
github won't let me comment on the exact lines because it considers them "outside" of the diff. Anyway, docs on line 136 and 137 (new code) should be updated. Also, I wonder if we should make it more clear that changing the type signature of this in a subclass is bad/unsupported?